### DISCUSSION TITLE
[bug] Deadwood doesn't resolve s3.wasabisys.com

### DISCUSSION THREAD

**Discussion by RalMay** [samboy/MaraDNS#125] [Q&A] [answered]
Hi,
found no "issues" page or other place to put this:
**Deadwood** (of MaraDNS **3-5-0036-win32**) runs pretty well on my **Win10x64** but is unable to resolve 1 single domain (**s3.wasabisys.com** - bc. of too much output?).

pls see screenshot
1 - no answer from Deadwood when looking up s3.wasabisys.com
2 - answer from Deadwood when looking up e.g. ibm.com
3 - answer of Deadwoods upstream server when looking up s3.wasabisys.com

The hopefully relevant part of _dwood3rc.txt_:
upstream_servers = {}
upstream_servers["."]="195.186.1.111, 195.186.1.110" # Servers we connect to

It would be nice if you could look into it or direct me to the right place to put this.
Thank you!

![1](https://github.com/samboy/MaraDNS/assets/65014385/81e3e04a-f006-4142-8fa4-836e493fab08)

**Comment by samboy** [comment]
Works for me:

```
$ askmara As3.wasabisys.com. 127.10.10.1
# Querying the server with the IP 127.10.10.1
# Question: As3.wasabisys.com.
s3.wasabisys.com. +120 cname wasabisys.com.
#wasabisys.com. +120 a 38.27.106.101
#wasabisys.com. +120 a 38.27.106.19
#wasabisys.com. +120 a 38.27.106.29
#wasabisys.com. +120 a 38.27.106.14
#wasabisys.com. +120 a 38.27.106.12
#wasabisys.com. +120 a 38.27.106.36
#wasabisys.com. +120 a 38.27.106.37
#wasabisys.com. +120 a 38.27.106.106
#wasabisys.com. +120 a 38.27.106.23
#wasabisys.com. +120 a 38.27.106.13
#wasabisys.com. +120 a 38.27.106.24
#wasabisys.com. +120 a 38.27.106.100
#wasabisys.com. +120 a 38.27.106.107
#wasabisys.com. +120 a 38.27.106.35
#wasabisys.com. +120 a 38.27.106.30
#wasabisys.com. +120 a 38.27.106.33
#wasabisys.com. +120 a 38.27.106.15
#wasabisys.com. +120 a 38.27.106.102
#wasabisys.com. +120 a 38.27.106.32
#wasabisys.com. +120 a 38.27.106.103
#wasabisys.com. +120 a 38.27.106.22
#wasabisys.com. +120 a 38.27.106.16
#wasabisys.com. +120 a 38.27.106.21
#wasabisys.com. +120 a 38.27.106.27
#wasabisys.com. +120 a 38.27.106.31
#wasabisys.com. +120 a 38.27.106.34
#wasabisys.com. +120 a 38.27.106.25
#wasabisys.com. +120 a 38.27.106.26
# NS replies:
# AR replies:
```

This is using the default upstream servers, e.g. 9.9.9.9 and 149.112.112.112.

**Reply by RalMay** [reply]
Works now on my system as well, didn't for at least a week.
Answer time is still pretty slow (~2s, just a bit more might have caused the timeout), same request from my upstream appears immediately.
Interestingly using your upstreams Deadwood answers (with a slow delay but) within <<1s.

Also Deadwood appears not to cache the request, each repeated nslookup takes these ~2s time (otherwise I used Quad9 as workaround to put the record in cache)?

**Comment by RalMay** [comment]
RN it again doesn't work and, looking at a [Wireshark capture](https://drive.filen.io/d/7526f01e-c6fb-4e4f-9111-6a50951c57dd#L6h4pypttiM5cZchRbQ1HOeetS9l2zfj)* it's clearly an issue with Deadwood as the upstream answered within 100ms (and from what I see in Wiresharks packet detail window (`deadwood-problem.pcapng` linked above) with valid values):

\* edit: exchanged the file for a test with 9.9.9.9 and 149.112.112.112 which shows the very same result
edit2: Same request directly to 9.9.9.9, apparently nslookup switched to TCP to actually resolve, pls. see [direct_request_to_9.9.9.9.pcapng](https://drive.filen.io/d/67f56ddd-180c-4528-b7a6-b22b2701297a#Qcky2MJncNh1oSI6AvKe9n5uWJ8hQUUh).

![1](https://github.com/samboy/MaraDNS/assets/65014385/e2541c93-fbf9-4acb-86c1-9ec1af1c3348)

**Reply by samboy** [reply, marked as answer]
Deadwood will not resolve answers which require DNS-over-TCP to resolve, and corner cases like this, where we can not get a meaningful packet over UDP, are not resolvable by Deadwood at this time.

```
$ askmara As3.wasabisys.com. 127.10.10.1
# Querying the server with the IP 127.10.10.1
# Remote server said: SERVER FAILURE
# Question: As3.wasabisys.com.
# NS replies:
# AR replies:

$ askmara As3.wasabisys.com. 9.9.9.9
# Querying the server with the IP 9.9.9.9
# Remote server said: TRUNCATED
# Question: As3.wasabisys.com.
s3.wasabisys.com. +107 cname wasabisys.com.
# NS replies:
# AR replies:

$ askmara Awasabisys.com. 9.9.9.9
# Querying the server with the IP 9.9.9.9
# Remote server said: TRUNCATED
# Question: Awasabisys.com.
# NS replies:
# AR replies:

$ askmara Awasabisys.com. 8.8.8.8
# Querying the server with the IP 8.8.8.8
# Remote server said: TRUNCATED
# Question: Awasabisys.com.
# NS replies:
# AR replies:

$ askmara As3.wasabisys.com. 8.8.8.8
# Querying the server with the IP 8.8.8.8
# Remote server said: TRUNCATED
# Question: As3.wasabisys.com.
# NS replies:
# AR replies:
```

**Comment by samboy** [comment]
Here is a verbose output of Deadwood’s log when it sees this:

```
Tuesday, March 26, 2024 1:44:31 AM: Got DNS query for \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:31 AM: Looking in cache for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:31 AM: Nothing found for \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:31 AM: Making connection to IP 149.112.112.112
Tuesday, March 26, 2024 1:44:31 AM: Caching a reply for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:32 AM: Connection for query \002s3\011wasabisys\003com\000\000\001 did not respond; trying again
Tuesday, March 26, 2024 1:44:32 AM: Making connection to IP 9.9.9.9
Tuesday, March 26, 2024 1:44:32 AM: Caching a reply for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:33 AM: Connection for query \002s3\011wasabisys\003com\000\000\001 did not respond; trying again
Tuesday, March 26, 2024 1:44:33 AM: Making connection to IP 149.112.112.112
Tuesday, March 26, 2024 1:44:33 AM: Caching a reply for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:34 AM: Connection for query \002s3\011wasabisys\003com\000\000\001 did not respond; trying again
Tuesday, March 26, 2024 1:44:34 AM: Making connection to IP 9.9.9.9
Tuesday, March 26, 2024 1:44:34 AM: Caching a reply for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:35 AM: Connection for query \002s3\011wasabisys\003com\000\000\001 did not respond; trying again
Tuesday, March 26, 2024 1:44:35 AM: Making connection to IP 149.112.112.112
Tuesday, March 26, 2024 1:44:35 AM: Caching a reply for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:36 AM: Connection for query \002s3\011wasabisys\003com\000\000\001 did not respond; trying again
Tuesday, March 26, 2024 1:44:36 AM: Making connection to IP 9.9.9.9
Tuesday, March 26, 2024 1:44:36 AM: Caching a reply for query \002s3\011wasabisys\003com\000\000\001
Tuesday, March 26, 2024 1:44:37 AM: Sending SERVER FAIL for query \002s3\011wasabisys\003com\000\000\001
```

I will close this but will make a mental note that having some level of EDNS support so Deadwood can receive larger packets would be a nice to have feature. We can safely receive packets up to 1280 bytes in size but it’s a question of making an EDNS0 packet which only requests a larger packet and doesn’t request DNSSEC packets.
